### PR TITLE
fix: Add Picture-in-Picture functionality to video player

### LINKED ISSUE #1295: bug: PIP(picture in picture) mode not available after the update
After the latest update the picture in picture mode has been removed due to which I am facing difficulties to code along Harkirat sir. The option to download the notes has also been removed. Please help i am following cohort 2 offline recorded videos.

**Expected behavior**
I really want pip(picture in picture) mode for cohort 2 offline recorded videos and it will be good if you fix the notes download related issues.

**Info (please complete the following information):**
 - Browser - chrome
 - Version - 128.0.6613.147

### PR DESCRIPTION
**Pull request by gupta-soham**
### PR Fixes:
- Implements Picture-in-Picture (PiP) mode for the video player
- Introduces `P` key shortcut for quick PiP mode toggle
- Expands playback rate options to include 2.5x and 3x speeds

Resolves #1295 

### Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I assure there is no similar/duplicate pull request regarding same issue

## Preview

https://github.com/user-attachments/assets/135bfa2c-7bfc-40aa-a1e0-f23c2e20ca07

### CODE REVIEW

**Comment by gupta-soham** [comment]
@hkirat @siinghd @devsargam 
Please have a look and review this PR.

File: src/components/VideoPlayer2.tsx [hunk 1]
```diff
@@ -331,7 +386,7 @@ export const VideoPlayer: FunctionComponent<VideoPlayerProps> = ({
         videoElement,
         {
           ...options,
-          playbackRates: [0.5, 1, 1.25, 1.5, 1.75, 2],
+          playbackRates: [0.5, 1, 1.25, 1.5, 1.75, 2, 2.5, 3],
```
Comment (devsargam) [inline src/components/VideoPlayer2.tsx:389]: can u remove 2.5 and 3 playback speed?

File: src/components/VideoPlayer2.tsx [hunk 2]
```diff
@@ -24,7 +27,7 @@ interface VideoPlayerProps {
   onVideoEnd: () => void;
 }
 
-const PLAYBACK_RATES: number[] = [0.5, 1, 1.25, 1.5, 1.75, 2];
+const PLAYBACK_RATES: number[] = [0.5, 1, 1.25, 1.5, 1.75, 2, 2.5, 3];
```
Comment (devsargam) [inline src/components/VideoPlayer2.tsx:30]: same as above

**Review by devsargam — CHANGES_REQUESTED** [review]

**Comment by devsargam** [comment]
Thank you sir

**Pull request merged**